packages/ymax-planner/src/engine.ts

```diff
@@ -347,18 +373,78 @@ export const startEngine = async ({
 
   // TODO: verify consumption of paginated data.
   const { query } = signingSmartWalletKit;
-  const portfolioKeys = await query.vstorage.keys(VSTORAGE_PATH_PREFIX);
+  const portfolioKeys = await query.vstorage.keys(PORTFOLIOS_PATH_PREFIX);
   const portfolioKeyForDepositAddr = new Map() as Map<Bech32Address, string>;
   await makeWorkPool(portfolioKeys, undefined, async portfolioKey => {
     const status = await query.readPublished(
-      `${stripPrefix('published.', VSTORAGE_PATH_PREFIX)}.${portfolioKey}`,
+      `${stripPrefix('published.', PORTFOLIOS_PATH_PREFIX)}.${portfolioKey}`,
     );
     mustMatch(status, PortfolioStatusShapeExt, portfolioKey);
     const { depositAddress } = status;
     if (!depositAddress) return;
     portfolioKeyForDepositAddr.set(depositAddress, portfolioKey);
   }).done;
 
+  const subscriptionKeys = await query.vstorage.keys(
+    TX_SUBSCRIPTIONS_PATH_PREFIX,
+  );
+  console.warn(
+    `Found ${subscriptionKeys.length} existing subscriptions to monitor`,
+  );
+
+  // Process existing pending subscriptions on startup
+  await makeWorkPool(subscriptionKeys, undefined, async subscriptionKey => {
+    const logIgnoredError = err => {
+      const msg = `⚠️ Failed to process existing subscription: ${subscriptionKey}`;
+      console.error(msg, err);
+    };
+
+    const [vstorageSettlement] = await Promise.allSettled([
```

Comment:
Clever, but inefficient and obscure. How about normal control flow?
```suggestion
    await null;
    let subscriptionData;
    try {
      // eslint-disable-next-line @jessie.js/safe-await-separator
      subscriptionData = await query.readPublished(
        stripPrefix(
          'published.',
          `${TX_SUBSCRIPTIONS_PATH_PREFIX}.${subscriptionKey}`,
        ),
      );
    } catch (err) {
      logIgnoredError(err);
      return;
    }
```